From 63bc19d7c3a2efced171c72fbcc49b5e06c4764c Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:02:08 -0400
Subject: [ethernet] mlx4: do not ignore autoneg in mlx4_en_set_pauseparam()

Message-id: <157c60fed32b68b6b407180dd1bd42f6f95e7dfa.1410425016.git.avadai@redhat.com>
Patchwork-id: 93759
O-Subject: [Patch RHEL7.1 V1 192/197] net/mlx4_en: do not ignore autoneg in mlx4_en_set_pauseparam()
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Ivan Vecera <ivecera@redhat.com>

The driver does not support pause autonegotiation so it should return
-EINVAL when the function is called with non-zero autoneg.

Cc: Amir Vadai <amirv@mellanox.com>
Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 278d436a476f69fc95d5c82bf61b6c2d02f4d44e)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c b/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
index 245f3b7..7979f98 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
@@ -487,6 +487,9 @@ static int mlx4_en_set_pauseparam(struct net_device *dev,
  struct mlx4_en_dev *mdev = priv->mdev;
  int err;
 
+ if (pause->autoneg)
+  return -EINVAL;
+
  priv->prof->tx_pause = pause->tx_pause != 0;
  priv->prof->rx_pause = pause->rx_pause != 0;
  err = mlx4_SET_PORT_general(mdev->dev, priv->port,
-- 
1.7.1